src/main/java/com/gachigang/ontherun/service/AuthService.java

```diff
@@ -36,12 +55,75 @@ public void authenticateUser(LoginRequest loginRequest) {
         SecurityContextHolder.getContext().setAuthentication(authentication);
     }
 
-    public User register(RegisterRequest registerRequest) {
+    public AuthenticationResponse register(RegisterRequest registerRequest) {
+        Role role = roleRepository.findById(registerRequest.getRoleId())
+                .orElse(roleRepository.findByName(UserRole.USER.getRole()));
+
         User user = User.builder()
                 .email(registerRequest.getEmail())
                 .password(passwordEncoder.encode(registerRequest.getPassword()))
+                .roles(Collections.singleton(role))
+                .build();
+
+        User savedUser = userRepository.save(user);
+        String jwtToken = jwtService.generateToken(user);
+        String refreshToken = jwtService.generateRefreshToken(user);
+        saveUserToken(savedUser, jwtToken);
+
+        return AuthenticationResponse.builder()
+                .accessToken(jwtToken)
+                .refreshToken(refreshToken)
+                .build();
+    }
+
+    private void saveUserToken(User user, String jwtToken) {
```

Comment:
Seems like it should be public method in TokenService, what do you think?

Reply:
I guess, it's better to keep it private because it won't be used anywhere in the future.